src/tpm/apple_secure_enclave/key_handle.rs

```diff
@@ -116,3 +118,13 @@ impl fmt::Debug for AppleSecureEnclaveKeyPair {
             .finish()
     }
 }
+
+impl Drop for AppleSecureEnclaveKeyPair {
+    fn drop(&mut self) {
+        if self.storage_manager.is_none() {
```

Comment:
An ephemeral key can be created with `self.storage_manager.is_some()`. In this case the ephemeral key is never deleted.